src/target/riscv/riscv-013.c

```diff
@@ -2426,130 +2426,95 @@ static int init_target(struct command_context *cmd_ctx,
 static int assert_reset(struct target *target)
 {
 	RISCV013_INFO(info);
+	int result;
 
 	select_dmi(target);
 
-	uint32_t control_base = set_field(0, DM_DMCONTROL_DMACTIVE, 1);
-
 	if (target_has_event_action(target, TARGET_EVENT_RESET_ASSERT)) {
 		/* Run the user-supplied script if there is one. */
 		target_handle_event(target, TARGET_EVENT_RESET_ASSERT);
-	} else if (target->rtos) {
-		/* There's only one target, and OpenOCD thinks each hart is a thread.
-		 * We must reset them all. */
-
-		/* TODO: Try to use hasel in dmcontrol */
-
-		/* Set haltreq for each hart. */
-		uint32_t control = control_base;
-
-		control = set_dmcontrol_hartsel(control_base, info->index);
-		control = set_field(control, DM_DMCONTROL_HALTREQ,
-				target->reset_halt ? 1 : 0);
-		dmi_write(target, DM_DMCONTROL, control);
-
-		/* Assert ndmreset */
-		control = set_field(control, DM_DMCONTROL_NDMRESET, 1);
-		dmi_write(target, DM_DMCONTROL, control);
-
 	} else {
-		/* Reset just this hart. */
-		uint32_t control = set_dmcontrol_hartsel(control_base, info->index);
+		uint32_t control = set_field(0, DM_DMCONTROL_DMACTIVE, 1);
+		control = set_dmcontrol_hartsel(control, info->index);
 		control = set_field(control, DM_DMCONTROL_HALTREQ,
 				target->reset_halt ? 1 : 0);
 		control = set_field(control, DM_DMCONTROL_NDMRESET, 1);
-		dmi_write(target, DM_DMCONTROL, control);
+		result = dmi_write(target, DM_DMCONTROL, control);
+		if (result != ERROR_OK)
+			return result;
 	}
 
 	target->state = TARGET_RESET;
 
-	dm013_info_t *dm = get_dm(target);
-	if (!dm)
-		return ERROR_FAIL;
-
 	/* The DM might have gotten reset if OpenOCD called us in some reset that
 	 * involves SRST being toggled. So clear our cache which may be out of
 	 * date. */
-	riscv013_invalidate_cached_debug_buffer(target);
-
-	return ERROR_OK;
+	return riscv013_invalidate_cached_debug_buffer(target);
 }
 
 static int deassert_reset(struct target *target)
 {
 	RISCV013_INFO(info);
-	select_dmi(target);
+	int result;
 
+	select_dmi(target);
 	/* Clear the reset, but make sure haltreq is still set */
 	uint32_t control = 0, control_haltreq;
 	control = set_field(control, DM_DMCONTROL_DMACTIVE, 1);
 	control_haltreq = set_field(control, DM_DMCONTROL_HALTREQ, target->reset_halt ? 1 : 0);
-	dmi_write(target, DM_DMCONTROL,
+	result = dmi_write(target, DM_DMCONTROL,
 			set_dmcontrol_hartsel(control_haltreq, info->index));
+	if (result != ERROR_OK)
+		return result;
 
 	uint32_t dmstatus;
-	int dmi_busy_delay = info->dmi_busy_delay;
+	const int dmi_busy_delay = info->dmi_busy_delay;
```

Comment:
For readability, could we please call this this `orig_dmi_busy_delay`?

```suggestion
	const int orig_dmi_busy_delay = info->dmi_busy_delay;
```

Reply:
Addressed.